src/patchelf.cc

```diff
@@ -741,10 +737,9 @@ void ElfFile<ElfFileParamNames>::rewriteSectionsLibrary()
        since DYN executables tend to start at virtual address 0, so
        rewriteSectionsExecutable() won't work because it doesn't have
        any virtual address space to grow downwards into. */
-    if (isExecutable && startOffset > startPage) {
+    if (startOffset > startPage)
         debug("shifting new PT_LOAD segment by %d bytes to work around a Linux kernel bug\n", startOffset - startPage);
-        startPage = startOffset;
-    }
+    startPage = startOffset;
```

Comment:
I understand why `isExecutable` was removed as a conditional but not why `startPage = startOffset` was moved out.